### DISCUSSION TITLE
Socket emit is not fired when passing readAsBinaryString

### DISCUSSION THREAD

**Discussion by Tzvetelin88** [socketio/socket.io#4182] [Q&A] [answered]
Emitting FileReader data as "readAsDataURL", "readAsBinaryString" or other read type is not working on Socket.IO 4.4.0 / 4.3.0 / 3.3.0, but works fine in 2.2.0.

I'm using SocketIO to upload/download files. 
example client side - Angular:
```
ngOnInit() {
  // ....
     blobBlockSize: number = 3 * 1024 * 100// 100kb
     //The Variable that will hold the new Block of data
      let newFileBlob: Blob;

      // file.slice(loaded, loaded + step(5mb));
      newFileBlob = this.selectedFile.slice(
        startingRange,
        startingRange +
          Math.min(this.blobBlockSize, this.selectedFile.size - startingRange)
      );

      // Get next Blob portion of data to be written
      this.fReader.readAsArrayBuffer(newFileBlob);
  // ....
}

upload() {
    this.fReader = new FileReader();
    this.socket.emit("Upload", { fileName: this.name, data: evnt.target.result });
}
```

example server side - nodejs:
```
 socket.on("Upload", (data) => {
  console.log(`Upload`);    // It doesn't hit it
  //code here....
}
```

When I updated socket.io to 4.4.0 I notice we hit `this.socket.emit("Upload", { fileName: this.name, data: evnt.target.result });`, but it never fires `socket.on("Upload", ...)`
If I do `this.socket.emit("Upload", { fileName: this.name, data: 'just a test string'});`. it triggers fine.

**Comment by Tzvetelin88** [comment, marked as answer]
I found the issue to be size of buffer that I'm sending. 
I have tried first with 5MB, then 3MB, and found out it's working ok with less, ex. 100kb.
It seems there is some limitation applied on the newer versions

**Reply by darrachequesne** [reply]
Please check https://socket.io/docs/v4/server-options/#maxhttpbuffersize

The value was indeed lowered in v3, from 100 Mb to 1 MB.

Related: https://socket.io/docs/v4/troubleshooting-connection-issues/#you-are-trying-to-send-a-huge-payload
